### PR TITLE
fix: enforce code_hash is 32 bytes long

### PR DESCRIPTION
**Pull request by ClementWalter**
Solves: #188

### CODE REVIEW

**Comment by ClementWalter** [comment]
See 
image

for issue #188

File: cairo/ethereum/cancun/fork_types.cairo [hunk 1]
```diff
@@ -50,11 +50,10 @@ func encode_account{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, keccak_ptr: K
...
     let code_hash = keccak256(raw_account_data.value.code);
-    let code_hash_len = _encode_uint256_little(
-        dst + nonce_len + balance_len + storage_root_len, U256(code_hash.value)
-    );
-
-    let len = nonce_len + balance_len + storage_root_len + code_hash_len;
+    let code_hash_ptr = dst + nonce_len + balance_len + storage_root_len;
+    assert [code_hash_ptr] = 0x80 + 32;
+    uint256_to_bytes32_little(code_hash_ptr + 1, [code_hash.value]);
+    let len = nonce_len + balance_len + storage_root_len + 33;
```
[the remark this answers is not in the document]
Comment (obatirou) [reply cairo/ethereum/cancun/fork_types.cairo:56]: Oh right, RLP encoding

[the remark this answers is not in the document]
Comment (ClementWalter) [reply cairo/ethereum/cancun/fork_types.cairo]: added some comments regarding the hash encoding and magic numbers.

The general idea of everything inside `ethereum/**/*` is to refer to the reference python implementation

[the remark this answers is not in the document]
Comment (obatirou) [reply cairo/ethereum/cancun/fork_types.cairo]: Seems good then
Comment stands for magic numbers and documentation though

File: cairo/src/utils/bytes.cairo [hunk 2]
```diff
@@ -230,12 +230,18 @@ func uint256_to_bytes{range_check_ptr}(dst: felt*, n: Uint256) -> felt {
     return bytes_len;
 }
 
+func uint256_to_bytes32_little{range_check_ptr}(dst: felt*, n: Uint256) {
+    alloc_locals;
```
Comment (obatirou) [inline cairo/src/utils/bytes.cairo:234]: might not be needed

**Review by obatirou — APPROVED** [review]

**Pull request merged**